pong.h

```diff
@@ -12,12 +12,36 @@ namespace pong
 class Pong
 {
 private:
-    std::vector<std::unique_ptr<GameObject>>mGameObjects {};
+    Pong() = default;
+    Pong(const Pong&) = delete;
+    Pong& operator=(const Pong&) = delete;
+    Pong(Pong&&) = delete;
+    Pong& operator=(Pong&&) = delete;
+    ~Pong() = default;
+
+    std::vector<std::shared_ptr<GameObject>>mGameObjects {};
     CollisionManager mCollisionManager {};
 
 public:
-    void PongInit();
-    void PongGameLoop();
+    static Pong& GetInstance();
+
+    static void Init();
+    static void GameLoop();
+    static void Cleanup();
+
+    template<typename T>
+    static std::shared_ptr<T> FindGameObject()
```

Comment:
The only way I was able to make this function work was by implementing it fully in the header file. I am guessing it is some weird template issue?

Reply:
yep correct! Templates need to live in the header